server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -523,7 +529,17 @@ public TSExecuteStatementResp executeStatement(TSExecuteStatementReq req) {
       PhysicalPlan physicalPlan;
       physicalPlan = processor.parseSQLToPhysicalPlan(statement, zoneIds.get());
       if (physicalPlan.isQuery()) {
-        return executeQueryStatement(statement, physicalPlan);
+      	resp = executeQueryStatement(statement, physicalPlan);
+      	long endtime = System.currentTimeMillis();
+      	sqlArgument = new SqlArgument(resp,physicalPlan,statement,starttime,endtime);
```

Comment:
Duplicated.
Already pass `statement` as a parameter in constructor but the next line still invoke `setStatement(statement)`